zeq/publisher.h

```diff
@@ -10,24 +10,78 @@
 #include <zeq/api.h>
 #include <zeq/types.h>
 
+#include <memory>
+
 namespace zeq
 {
-namespace detail { class Publisher; }
 
 /**
  * Serves and publishes events, consumed by Subscriber.
  *
+ * The session is tied to ZeroConf announcement and can be disabled by passing
+ * zeq::NULL_SESSION as the session name.
+ *
  * Example: @include tests/publisher.cpp
  */
 class Publisher
 {
 public:
     /**
-     * Create a publisher on the given URI.
+     * Create a default publisher.
+     *
+     * Constrains:
```

Comment:
Constrain is a verb, the noun is constraint.
But there are more postconditions than constraints, aren't they?